cloudify_cli/tests/commands/test_agents.py

```diff
@@ -0,0 +1,226 @@
+########
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+from mock import MagicMock, patch
+
+from .test_base import CliCommandTest
+from cloudify_rest_client import deployments, tenants
+from cloudify_cli.exceptions import SuppressedCloudifyCliError
+from cloudify_rest_client.client import CLOUDIFY_TENANT_HEADER
+
+DEFAULT_TENANT_NAME = 'tenant0'
+
+
+class AgentsTests(CliCommandTest):
+    def setUp(self):
+        super(AgentsTests, self).setUp()
+        self.use_manager()
+
+    @staticmethod
+    def create_tenants_and_deployments(num_of_tenants, num_of_deps,
+                                       unique_deps_id=True):
+        tenants_list = []
+        deps = {}
+        index = 0
+        # create requested num of tenants
+        for i in range(num_of_tenants):
+            ten = tenants.Tenant({'name': 'tenant{0}'.format(i)})
+            tenants_list.append(ten)
+        # create requested num of deployments for each tenant
+        for tenant in tenants_list:
+            deps[tenant['name']] = []
+            if not unique_deps_id:
+                index = 0
+            for i in range(num_of_deps):
+                deps[tenant['name']].append(deployments.Deployment({
+                    'deployment_id': 'dep{0}'.format(index),
+                    'tenant_name': tenant['name']}))
+                index += 1
+
+        return tenants_list, deps
+
+    def mock_client(self, tenants, deps, nodes, deps_get):
+        def dep_list():
+            tenant_name = self.client._client.headers.get(
+                CLOUDIFY_TENANT_HEADER)
+            if not tenant_name:
+                tenant_name = DEFAULT_TENANT_NAME
+            return deps[tenant_name]
+        self.client.tenants.list = MagicMock(return_value=tenants)
+        self.client.deployments.list = dep_list
+        self.client.node_instances.list = MagicMock(return_value=nodes)
+        self.client.deployments.get = deps_get
+
+    @patch('cloudify_cli.commands.agents.run_worker')
+    def test_agents_install_multiple_tenants_and_deployments(self,
+                                                             worker_mock):
+        """
+        Expected behavior: install agents for all deployments in current
+                           tenant.
+        """
+        tenants_list, deps = AgentsTests.create_tenants_and_deployments(2, 3)
```

Comment:
You can do `self.create_tenants_and_deployments`, even though it's a static method